Automatic code generation for choreo (#1298)
SleipnirGroup/Choreo: docs/usage/code-generation.md

> # Code Generation(Java Only)

Reviewer: ```suggestion
# Code Generation (Java Only)
```

> # Code Generation (Java Only)

Reviewer 2: Needs to be updated

To enable or disable this feature, simply go to the "Code Generation" tab in the document settings.

!!! tip

    These generated files do not depend on ChoreoLib to work. They will be perfectly compatible with any java project.

## Variables

> Choreo can automatically generate java files containing variables defined in the Choreo GUI. This allows information like pre-defined poses and robot characteristics to be shared between robot code and the GUI, while ensuring consistency. Values will use the java units library when possible.

Reviewer: ```suggestion
Choreo can automatically generate Java files containing variables defined in the Choreo GUI. This allows information such as predefined poses and robot characteristics to be shared between the robot code and the GUI, while ensuring consistency. Values will use the Java units library when possible.
```

## Trajectory Names

> Choreo can also generate a file containing the names of all created trajectories. Each trajectory name is mapped to a static constant within the file. By referencing these static constants in place of strings, the java compiler can help you detect references to mispelled(or nonexistent) trajectories.

Reviewer: ```suggestion
Choreo can also generate a file listing all created trajectories. Each trajectory name is mapped to a static constant within the file. By referencing these static constants in place of strings, the Java compiler can help you detect references to misspelled (or non-existent) trajectories.
```
